.gdbinit

```diff
@@ -651,5 +754,124 @@ end
 
 document lookup_root
 	lookup a refcounted in root
-	usage: lookup_root [ptr].
+	Usage: lookup_root [ptr].
+end
+
+#### Functions to dump opline information ####
+# TODO: print more meaningful information
+# arg0: (zend_op*)
+define dump_opline
+	if $arg0
+		set $opline=$arg0
+		printf "lineno=[%d] opcode=[%d] ", $opline->lineno, $opline->opcode
+		printf "handler=["
+		output $opline->handler
+		printf "]\n"
+	else
+		printf "[ERROR] Invalid zend_op* argument is NULL\n"
+	end
+
+end
+
+document dump_opline
+	Print opline information
+	Usage: dump_opline (zend_op*)opline
+end
+
+define dump_current_opline
+	____executor_globals
+	if $eg.current_execute_data
+		dump_opline $eg.current_execute_data->opline
+	else
+		printf "[ERROR] $eg.current_execute_data is NULL\n"
+	end
+end
+
+document dump_current_opline
+	Dump currently executed opline information
+	Usage: print_executed_opline
+end
+
+# arg0: (zend_op_array*)
+define dump_op_array
```

Comment:
I wonder if we can't just re-use `zend_dump_op_array`. That is only available when compiling with opcache though.

Reply:
Sorry. I did not quite catch the meaning... 
`zend_dump_op_array` is a c function, can we call it in gdb?
`dump_op_array` just let us check an op_array at any time.

Reply:
Yeah nevermind I don't think that will work well.